test/unit/variable_test.rb

```diff
@@ -232,6 +232,59 @@ def test_filter_argument_error_translation
     assert_equal('Liquid error: wrong number of arguments (given 1, expected 2)', exc.message)
   end
 
+  class IntegerDrop < Liquid::Drop
+    def initialize(value)
+      super()
+      @value = value.to_i
+    end
+
+    def to_liquid_value
+      @value
+    end
+  end
+
+  def test_to_liquid_value_on_variable_lookup
+    context = {
+      'number' => IntegerDrop.new('1'),
+      'list' => [1, 2, 3, 4, 5],
+    }
+
+    output = variable_strict_parse('list[number]').render!(context)
+    assert_equal('2', output)
+  end
+
+  class BooleanDrop < Liquid::Drop
+    def initialize(value)
+      super()
+      @value = value
+    end
+
+    def to_liquid_value
+      @value
+    end
+
+    def ==(other)
+      other == @value
+    end
+
+    def to_s
+      @value ? 'Yay' : 'Nay'
+    end
+  end
+
+  def test_to_liquid_value_on_variable_lookup_and_if_statement
```

Comment:
This isn't liquid-c specific, so it belongs in the liquid repo under the test/integration/ directory and it will get tested with liquid-c